### DISCUSSION TITLE
swanctl --log: Debug a specific connection?

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#2534] [Q&A]
Hi,

how can I debug the behavior of a specific connection with swanctl --log? With quite a number of tunnels the log is quite noisy. Is it possible to select a specific connection for the log output?

Something like swanctl --log --ike my_connection

**Comment by tobiasbrunner** [comment]
You can't. In particular because that command does not allow debugging in the first place (it only supports log level 1 to avoid loops). However, if you subscribe to the `log` event via [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) directly, and the log message happens in the context of an IKE_SA, you could filter messages via `ikesa-name` attribute (similar to enabling `ike_name` for [other loggers](https://docs.strongswan.org/docs/5.9/config/logging.html)).  I guess such a filter could be added to the `--log` command as well.

What exactly do you want to "debug" this way?
